kotest-property/src/commonMain/kotlin/io/kotest/property/Gen.kt

```diff
@@ -59,42 +65,42 @@ sealed class Gen<out A> {
 }
 
 /**
- * An [Arb] (for arbitrary) is a provider of property test data in two categories: edgecases and samples.
+ * An [Arb] (short for arbitrary) is a generator of values in two categories: edgecases and samples.
  *
- * Edge cases are values that are a common source of bugs. For example, a function using ints is
+ * Edgecases are values that are a common source of bugs. For example, a function using ints is
  * more likely to fail for common edge cases like zero, minus 1, positive 1, [Int.MAX_VALUE] and [Int.MIN_VALUE]
- * rather than random values like 965489. Therefore, edge cases can be included in sequences
- * generated by an arbitrary.
+ * rather than random values like 965489. Therefore it is useful that we try to include such values
+ * rather than relying entirely on random values which are unlikely to generate these.
  *
  * Not all arbitraries will utilize edge cases. For example, if you define an integer generator
- * using a subset of the number space - say from 100 to 250,000 - then no edge cases are provided.
+ * using a subset of the number space - say from 100 to 200 - then no edge cases would be provided.
  *
  * Samples are chosen randomly from the sample space and are used to give a greater breadth to
  * the test cases. For example, in the case of a function using integers, these random values
  * could be from across the entire integer number line, or could be limited to a subset of ints
  * such as natural numbers or even numbers.
  *
- * In order to use an [Arb] inside a property test, one must invoke the [take] method, passing in the
- * number of iterations required and optionally a [ShrinkingMode].
+ * In order to use an [Arb] outside a property test, one must invoke the [take] method, passing in
+ * the number of iterations required and optionally a [ShrinkingMode].
  */
 abstract class Arb<out A> : Gen<A>() {
 
    /**
-    * Edgecase values for this type A.
+    * Returns a single edgecase for this arbitrary.
+    * If this arb provides mutliple edgecases, then one should be chosen randomly.
+    * Can return null if no edgecases are available.
     */
-   abstract fun edgecases(): List<A>
-
-   internal open fun edges(): Option<Edgecase<A>> =
-      NonEmptyList
-         .fromList(edgecases())
-         .map { Edgecase { rs, _ -> it.all.random(rs.random) } }
+   abstract fun edgecase(rs: RandomSource): A?
 
    /**
     * Returns a random [Sample] from this [Arb] using the supplied random source.
     */
    open fun sample(rs: RandomSource): Sample<A> = values(rs).first()
 
-   @Deprecated("implement one value at a time using sample(rs). This function will be removed in 4.6", ReplaceWith("sample(rs)"))
+   @Deprecated(
+      "implement one value at a time using sample(rs). This function will be removed in 4.7",
```

Comment:
We've been bumping the deprecation schedule for this..

Reply:
Yes, I intend 4.6 to be a very small release that has compatibility for Kotlin 1.5 as the only potentially breaking feature. 4.6 will come out pretty much as soon as JB release 1.5 so I don't expect there to be much content in it. So 4.7 can be the one were we come in and clean the house on the prop test framework.